src/backend/base/langflow/components/embeddings/HuggingFaceInferenceAPIEmbeddings.py

```diff
@@ -1,32 +1,59 @@
 from langchain_community.embeddings.huggingface import HuggingFaceInferenceAPIEmbeddings
 from pydantic.v1.types import SecretStr
+import requests
+from urllib.parse import urlparse
 
 from langflow.base.models.model import LCModelComponent
 from langflow.field_typing import Embeddings
 from langflow.io import MessageTextInput, Output, SecretStrInput
 
 
 class HuggingFaceInferenceAPIEmbeddingsComponent(LCModelComponent):
-    display_name = "HuggingFace Embeddings"
-    description = "Generate embeddings using Hugging Face Inference API models."
-    documentation = "https://github.com/huggingface/text-embeddings-inference"
+    display_name = "HuggingFace Embeddings Inference"
+    description = "Generate embeddings using HuggingFace Text Embeddings Inference (TEI)"
+    documentation = "https://huggingface.co/docs/text-embeddings-inference/en/index"
     icon = "HuggingFace"
     name = "HuggingFaceInferenceAPIEmbeddings"
 
     inputs = [
-        SecretStrInput(name="api_key", display_name="API Key"),
-        MessageTextInput(name="api_url", display_name="API URL", advanced=True, value="http://localhost:8080"),
+        SecretStrInput(
+            name="api_key", display_name="API Key", advanced=True, info="The API key is required for non-local API URLs"
```

Comment:
Maybe something like: `Required for non-local API URLs. Local inference does not require an API Key". 

I don't know if this is true. Is there any way we can convey to users that the key is not necessary for local deployments (or even hosted deployments that don't require a key), but also the API requires a key so we pass in that dummy key? 

Seems like a lot to explain in an info, so maybe better to just leave as-is, now that I type all that.

Thoughts?

Reply:
We can go with "Required for non-local API URLs. Local inference does not require an API Key."